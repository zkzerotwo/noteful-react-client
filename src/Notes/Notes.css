.Notes {
    --Notes-height: 88px;
    background-color: #010d1e;
    color: #cacaca;
    height: var(--Notes-height);
    margin-bottom: 8px;
    max-width: 380px;
    padding: 8px 0 8px 8px;
    position: relative;
}

.Notes__delete {
    --Notes-button-height: 32px;
    background-color: #041a3a;
    border: 0;
    color: #bdbcbc;
    float: right;
    font-size: 12px;
    height: var(--Notes-button-height);
    position: relative;
    top: calc((var(--Notes-button-height) + 32px) * -1);
    transform-origin: right bottom;
    transform: rotate(-90deg);
    width: calc(var(--Notes-height));
}

.Notes__delete:hover {
    background-color: #bd0909;
    color: white;
    cursor: pointer;
    font-weight: bold;
}

.Notes__title {
    font-family: var(--custom-font);
    font-size: 20px;
    margin: 0;
    max-width: 80%;
    min-height: 24px;
}

.Notes__title a {
    color: #cacaca;
    display: block;
    text-decoration: none;
}

.Notes__title a:hover {
    color: #FFB72F;
}

.Notes__dates {
    bottom: 8px;
    color: #a7a5a5;
    font-size: 14px;
    position: absolute;
}