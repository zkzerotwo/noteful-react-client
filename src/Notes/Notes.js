import React from 'react'
import { Link } from 'react-router-dom'
import NotefulContext from '../NotefulContext'
import PropTypes from 'prop-types'
import config from '../config'
import './Notes.css'
// import format from 'date-fns/format'


export default class Notes extends React.Component {
    static defaultProps = {
        onDeleteNote: () => {},
    }
    
    static contextType = NotefulContext;
    handleClickDelete = e => {
        e.preventDefault()
        const noteId = this.props.id
        console.log(noteId)
        fetch(`${config.API_ENDPOINT}/notes/${noteId}`, {
            method: 'DELETE',
            headers: {
                'content-type': 'application/json'
            },
        })
            .then(res => {
                if (!res.ok)
                    return res.json().then(e => Promise.reject(e))
            })
            .then(() => {
                this.context.deleteNote(noteId)
                this.props.onDeleteNote(noteId)
            })
            .catch(error => {
                console.error({ error })
            })
    }
    render() {
        const { name, id, modified } = this.props;
        console.log(this.props)
        return (
            <div className='Notes'>
                <h2 className='Notes__title'>
                    <Link to={`/note/${id}`}>
                        {name}
                    </Link>
                </h2>
                <button 
                className='Notes__delete' 
                type='button' 
                onClick={this.handleClickDelete}>
                    {' '}
        remove
      </button>
                <div className='Notes__dates'>
                    <div className='Notes__dates-modified'>
                        Modified
          {' '}
                        <span className='Date'>
                            {/* {format(new Date(modified), "uuuu MM do XXXXX ")} */}
                            {modified}
                        </span>
                    </div>
                </div>
            </div>
        )
    }
}

Notes.propTypes = {
    modified: PropTypes.string,
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
}