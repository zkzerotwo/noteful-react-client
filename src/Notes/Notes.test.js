import React from 'react';
import { shallow } from 'enzyme'
import  Enzyme  from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Notes from './Notes'

Enzyme.configure({ adapter: new Adapter() });
describe(`Note component`, () => {
    const props = {
      id: 'a',
      name: 'test-class-name',
      modified: new Date(2018, 12, 15),
    }
  
    it('renders a .Note by default', () => {
      const wrapper = shallow(<Notes />)
      expect(toJson(wrapper)).toMatchSnapshot()
    })
  
    it('renders the Note given props', () => {
      const wrapper = shallow(<Notes {...props} />)
      expect(toJson(wrapper)).toMatchSnapshot()
    })
  })
