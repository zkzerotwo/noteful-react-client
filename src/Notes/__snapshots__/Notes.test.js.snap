// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note component renders a .Note by default 1`] = `
<div
  className="Notes"
>
  <h2
    className="Notes__title"
  >
    <Link
      to="/notes/undefined"
    />
  </h2>
  <button
    className="Notes__delete"
    onClick={[Function]}
    type="button"
  >
     
    remove
  </button>
  <div
    className="Notes__dates"
  >
    <div
      className="Notes__dates-modified"
    >
      Modified
       
      <span
        className="Date"
      />
    </div>
  </div>
</div>
`;

exports[`Note component renders the Note given props 1`] = `
<div
  className="Notes"
>
  <h2
    className="Notes__title"
  >
    <Link
      to="/notes/a"
    >
      test-class-name
    </Link>
  </h2>
  <button
    className="Notes__delete"
    onClick={[Function]}
    type="button"
  >
     
    remove
  </button>
  <div
    className="Notes__dates"
  >
    <div
      className="Notes__dates-modified"
    >
      Modified
       
      <span
        className="Date"
      >
        <Component />
      </span>
    </div>
  </div>
</div>
`;
